dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -1057,7 +1057,9 @@ public Workflow publish(@ApiParam(hidden = true) @Auth User user,
             // Create a SharedWorkFlow map for each Role and the list of workflows that belong to it
             final List<Workflow> workflows = workflowList.stream()
                 // Filter only the workflows that belong to the current Role and where the user is not the owner
-                .filter(workflow -> e.getValue().contains(workflow.getWorkflowPath()) && !workflow.getUsers().contains(user))
+                .filter(workflow -> e.getValue().contains(workflow.getWorkflowPath()))
+                // This causes a connection pool leak (the active connections keeps going up)
```

Comment:
This is a great find!

But I wonder what the underlying issue is? I am concerned that:

1. This breaks some functionality, although because we don't really have major shared workflows usage going now, it doesn't matter today. It might matter if the full integration with our partner happens.
2. I wonder if it happens in other contexts, i.e., other code iterating through a workflow's users might run into the same problem.